src/ioc.config.ts

```diff
@@ -12,21 +12,29 @@ import {Config} from './lib/util/Config';
 import {ProdOverrides, Services} from './Constants';
 
 function setupProd(): void {
-	container.register(Services.EnvOverridable, ProdOverrides);
+	// This method may be called more than once in unit test scenarios where
```

Comment:
The issue here is that any ioc container setup during an oclif unit test is then duplicated when the command runs. This path will only ever execute when running a test. Interested in potentially better solutions.

Reply:
Could you elaborate on this please? Could this lead to cases where tests flap, or where the ordering of tests within a file is important? Also, do you have any recommendations for how we'd go about fixing this?

Reply:
Ideally, we should never be calling this method from tests. Would it make sense to create a separate work item to fix any test that calls this?

Reply:
Both `setupProd` and `setupTestAlternatives` are currently called on the `release` branch when running a unit test. You can see this by adding logging to each method then running a test. This is the output of 

`>nyc mocha test/commands/scanner/rule/list.test.ts` on the release branch. 

`>nyc mocha test/commands/scanner/rule/list.test.ts
JB: setupProd called
JB: setupTestAlternatives called`

We could have the prod code skip this if it is running in the context of a unit test, but then the prod code knows about tests. It used to be that the prod code ran before the test code because it was triggered by any code that included `ioc.config.ts`, now the test code is running before the prod code, which is why the check is needed, it is only needed when specifically running the e2e scanner oclif tests.